File: alexandria/ui-elements/src/displays/components/Download.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractDownload from "../../../gen/displays/components/AbstractDownload";
import DownloadNotifier from "../../../gen/displays/notifiers/DownloadNotifier";
import DownloadRequester from "../../../gen/displays/requesters/DownloadRequester";
import Operation from "./Operation";

class Download extends AbstractDownload {

	constructor(props) {
		super(props);
		this.notifier = new DownloadNotifier(this);
		this.requester = new DownloadRequester(this);
	};

}

export default withStyles(Operation.Styles, { withTheme: true })(Download);

File: alexandria/ui/src/io/intino/alexandria/ui/displays/components/Download.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.events.operation.DownloadEvent;
import io.intino.alexandria.ui.displays.events.operation.DownloadListener;
import io.intino.alexandria.ui.displays.notifiers.DownloadNotifier;

import java.util.List;

public class Download<DN extends DownloadNotifier, B extends Box> extends AbstractDownload<DN, B> {
    private String option = null;
    private java.util.List<String> options;
    private DownloadListener downloadListener;

    public Download(B box) {
        super(box);
    }

    public List<String> options() {
        return options;
    }

    public Download<DN, B> options(List<String> options) {
        this.options = options;
        return this;
    }

    public Download<DN, B> select(String option) {
        this.option = option;
        return this;
    }

    public void changeSelection(String selection) {
        this.option = selection;
    }

    public void onExecute(DownloadListener listener) {
        this.downloadListener = listener;
    }

    @Override
    public void execute() {
        if (this.downloadListener == null) return;
        notifier.refreshResult(this.downloadListener.accept(new DownloadEvent(this, option)));
    }

}